src/native/fs/fs_napi.cpp

```diff
@@ -1744,6 +1747,41 @@ struct FileFlock : public FSWrapWorker<FileWrap>
     }
 };
 
+struct FileFcntlLock : public FSWrapWorker<FileWrap>
+{
+    struct flock fl;
+    FileFcntlLock(const Napi::CallbackInfo& info)
+        : FSWrapWorker<FileWrap>(info)
+        , fl()
+    {
+        // lock entire file
+        fl.l_whence = SEEK_SET;
+        fl.l_start = 0;
+        fl.l_len = 0;
+        fl.l_pid = 0;
+        fl.l_type = F_RDLCK;
+
+        if (info.Length() > 1 && !info[1].IsUndefined()) {
+            auto mode = info[1].As<Napi::String>().Utf8Value();
+            if (mode == "EXCLUSIVE") {
+                fl.l_type = F_WRLCK;
+            } else if (mode == "UNLOCK") {
+                fl.l_type = F_UNLCK;
+            } else {
```

Comment:
maybe better to also check explicitly for mode == "SHARE" and then else fail

Reply:
Yeah, that works as well. Will update the PR.

Reply:
please apply the same also to the existing flock code